### DISCUSSION TITLE
Schema validation for config file

### DISCUSSION THREAD

**Discussion by europ** [ansible/ansible-lint#837] [Ideas]
##### Summary

Schema of ansible-lint configuration file can be validated via [YAMALE tool](https://github.com/23andMe/Yamale), which also supports definitions for custom rules.

##### Issue Type

- Feature Idea

##### Additional Information
This issue has been opened in account to a discussion available here - https://github.com/ansible/ansible-lint/commit/7348a240ff19c0ea75f9bc26d1ff8b6c2cbab1ff#r39969016

Possible solution for [`TODO(ssbarnea): implement schema validation for config file`](https://github.com/ansible/ansible-lint/blob/master/lib/ansiblelint/cli.py#L73) in [`lib/ansiblelint/cli.py`](https://github.com/ansible/ansible-lint/blob/master/lib/ansiblelint/cli.py) in fuction `def load_config(config_file):`.

**Comment by webknjaz** [comment]
It'd be great to see a comparison with other solutions.

**Comment by gundalow** [comment]
I like the suggestion, thanks for raising it.

Just as a point of reference we use the Python voluptuous package in a few places in ansible-base's CI for validating different parts of the code base. I've no idea how that compares to YAMALE.
